controller/api/public/client.go

```diff
@@ -104,18 +106,32 @@ func (c *grpcOverHTTPClient) TapByResource(ctx context.Context, req *pb.TapByRes
 		return nil, err
 	}
 
-	if err := checkIfResponseHasError(httpRsp); err != nil {
-		httpRsp.Body.Close()
+	client, err := getStreamClient(ctx, httpRsp)
+	if err != nil {
 		return nil, err
 	}
 
-	go func() {
-		<-ctx.Done()
-		log.Debug("Closing response body after context marked as done")
-		httpRsp.Body.Close()
-	}()
+	return &tapClient{client}, nil
+}
 
-	return &tapClient{ctx: ctx, reader: bufio.NewReader(httpRsp.Body)}, nil
+func (c *grpcOverHTTPClient) Get(ctx context.Context, req *destinationPb.GetDestination, _ ...grpc.CallOption) (destinationPb.Destination_GetClient, error) {
```

Comment:
> unrelated: `Get` is unfortunately named for it to be shared across `grpcOverHTTPClient` like this. down the road we may want to rethink this whole interface anyway, maybe just like, vanilla gRPC?

Not sure I understood your point. `Get` is named so so that it conforms to the `DestinationClient`  protobuf interface. Changing its name would imply changing that contract, on which the proxy depends.

Reply:
sorry that comment around the naming of `Get` was not very actionable, nothing to change to here. it was more a general observation that we risk naming collisions when we have a struct like `grpcOverHTTPClient` supporting multiple interfaces.